lib/inc/cpp-pcp-client/connector/connector.hpp

```diff
@@ -59,6 +60,10 @@ class LIBCPP_PCP_CLIENT_EXPORT Connector {
               uint32_t pong_timeouts_before_retry = 3,
               long ws_pong_timeout_ms = 30000);
 
+    /// Calls stopMonitorTaskAndWait if the monitoring thread is
+    /// still active. This can result in throwing, so to safely handle
+    /// any exceptions that occured in the monitoring thread, first
+    /// call stopMonitoring and catch exceptions from that.
```

Comment:
I'm not sure about this. Maybe it's better to just log an error in that case. We can chat about this later.